### PR TITLE
Payec 186 review part 2

### PR DESCRIPTION
**Pull request by kastoestoramadus**
Yuan will copy the solution to the rest code and the PR will be ready

### CODE REVIEW

File: app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/TaxResult.scala [hunk 1]
```diff
@@ -38,42 +39,45 @@ object TaxResult {
...
     * buildTaxCalc will use the number returned to calculate the weekly gross pay from Daily or Hourly via those case classes.
     **/
-  def extractHours(quickCalcAggregateInput: QuickCalcAggregateInput): Int = quickCalcAggregateInput.savedSalary match {
-    case Some(s) => s.period match {
-      case "daily" => s.howManyAWeek.getOrElse(-1)
-      case "hourly" => s.howManyAWeek.getOrElse(-1)
+  private[quickmodel] def extractHours(quickCalcAggregateInput: QuickCalcAggregateInput): Int =
+    quickCalcAggregateInput.savedSalary match {
+      case Some(s) => s.period match {
+        case "daily" => s.howManyAWeek.getOrElse(-1)
```
Comment (kastoestoramadus) [inline app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/TaxResult.scala]: -1 seems to be never present during the journey flow. Maybe throwing an RuntimeException?

**Comment by kastoestoramadus** [comment]
Lets @JTY009 merge it into the master.

File: app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/UserTaxCode.scala [hunk 2]
```diff
@@ -69,16 +71,14 @@ object UserTaxCode extends TaxCalculatorHelper {
...
-        case Some(code) if code.gaveUsTaxCode => "checked"
-        case _ => ""
-      }
-    else
+    else {
       taxCode.value match {
-        case Some(code) if !code.gaveUsTaxCode && !taxCode.hasErrors => "checked"
+        case Some(code) if selection && code.gaveUsTaxCode ||
+          (!selection && (!code.gaveUsTaxCode && !taxCode.hasErrors)) =>
+          "checked"
```
Comment (kastoestoramadus) [inline app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/UserTaxCode.scala]: @YuanZHANG910 Check if it is logic equivalent.
Tests passed.

Comment (kastoestoramadus) [inline app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/UserTaxCode.scala]: Still something is too complicated in those conditions.

File: app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/UserTaxCode.scala [hunk 3]
```diff
@@ -35,50 +35,57 @@ object UserTaxCode extends TaxCalculatorHelper {
   implicit val format = Json.format[UserTaxCode]
 
   val defaultScottishTaxCode = "S1150L"
```
Comment (JTY009) [inline app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/UserTaxCode.scala]: Please change for consistency @YuanZHANG910

**Comment by YuanZHANG910** [comment]
done

File: app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/UserTaxCode.scala (hunk 3 above)
Comment (YuanZHANG910) [inline app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/UserTaxCode.scala]: done

**Pull request merged**